state/spaces.go

```diff
@@ -155,6 +155,25 @@ func (st *State) addSpaceTxnOps(
 	}
 	id := strconv.Itoa(seq)
 
+	ops := st.addSpaceTxnOps(id, name, providerId, isPublic)
+
+	for _, subnetId := range subnets {
+		// TODO:(mfoord) once we have refcounting for subnets we should
```

Comment:
I'm not sure about explicitly *storing* a ref-count. I'd prefer to work with derived data until there's a performance case for denormalisation.

This is how I am thinking about things now, although there are some serious implications/edge-cases that need to be thought through:
- Subnets get an ID, as spaces have (so we can have the same CIDR in different networks).
- Addresses get subnet IDs (validated with CIDR) instead of space names.
- Addresses are then in spaces by derivation and;
- Subnets can be determined to be in use by derivation.

A penny for @jameinel 's thoughts.